banyan/resource_service_web.go

```diff
@@ -254,12 +423,37 @@ func expandWebBackend(d *schema.ResourceData) (backend service.Backend) {
 	backend = service.Backend{
 		BackendTarget:       expandWebTarget(d),
 		ConnectorName:       d.Get("connector").(string),
-		BackendDNSOverrides: map[string]string{},
-		BackendWhitelist:    []string{},
+		BackendDNSOverrides: expandBackendDNSOverrides(d),
+		BackendWhitelist:    expandBackendWhitelist(d),
+		HttpConnect:         false,
 	}
 	return
 }
 
+func expandBackendWhitelist(d *schema.ResourceData) []string {
+	itemsRaw := d.Get("whitelist").([]interface{})
+	items := make([]string, len(itemsRaw))
+	for i, raw := range itemsRaw {
+		items[i] = raw.(string)
+	}
+	return items
+}
+
+func expandBackendDNSOverrides(d *schema.ResourceData) map[string]string {
+	dnsOverrides := make(map[string]string)
+	v, ok := d.GetOk("dns_overrides")
+
+	if !ok || len(v.(map[string]interface{})) <= 0 {
```

Comment:
Why `<= 0` instead of `== 0`? `len()` is always non-negative.

Reply:
sure @yoshiotu i should change to ==